bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/gems/mascot/MagLimits.scala

```diff
@@ -1,26 +1,27 @@
 package edu.gemini.ags.gems.mascot
 
+import edu.gemini.spModel.core.{Magnitude, MagnitudeBand}
+import scalaz._
+import Scalaz._
+
 /**
  * Defines limits for magnitude bands (Do we need upper and lower bounds?)
  */
-final class MagLimits(val bmag: Double,
-                      val vmag: Double,
-                      val rmag: Double,
-                      val jmag: Double,
-                      val hmag: Double,
-                      val kmag: Double) {
+case class MagLimits private (bmag: Double,
+                     vmag: Double,
+                     rmag: Double,
+                     jmag: Double,
+                     hmag: Double,
+                     kmag: Double) {
+
+  // preallocate to use in filter
+  private lazy val filters = List(new Magnitude(bmag, MagnitudeBand.B).some, new Magnitude(vmag, MagnitudeBand.V).some, new Magnitude(rmag, MagnitudeBand.R).some, new Magnitude(jmag, MagnitudeBand.J).some, new Magnitude(hmag, MagnitudeBand.H).some, new Magnitude(kmag, MagnitudeBand.K).some)
 
   /**
    * Returns true if the given star is within the mag limits
    */
-  def filter(star: Star): Boolean = {
-    star.bmag <= bmag &&
-      star.vmag <= vmag &&
-      star.rmag <= rmag &&
-      star.jmag <= jmag &&
-      star.hmag <= hmag &&
-      star.kmag <= kmag
-  }
+  def filter(star: Star): Boolean =
+    filters.map(f => star.target.magnitudeIn(f.get.band) <= f).map(Tags.Conjunction).suml
```

Comment:
Had to puzzle that out!  Assuming a `filters` list without the `Option` wrappers, is this the same as

````scala
filters.forall { m => star.target.magnitudeIn(m.band).exists(_.value <= m.value) }
````